source/slang/slang-options.cpp

```diff
@@ -632,7 +634,7 @@ struct OptionsParser
                     SLANG_RETURN_ON_FAIL(tryReadCommandLineArgument(sink, arg, &argCursor, argEnd, name));
                     addSharedLibraryPath(SharedLibraryType::Fxc, name);
                 }
-                else if (argStr[1] == 'D')
+                else if (argStr.getLength() >= 2 && argStr[1] == 'D')
```

Comment:
I *thought* we were already checking `argStr[0] == '-'` above, so that the check against length here shouldn't be needed, assuming null-terminated strings for all the arguments (which I think is fair...).

Reply:
True but it's accessed thru argStr which is a string and bounds checked.

Reply:
Ugh. When did *that* happen? :)

Pretty soon there aren't going to be any seat-of-our-pants raw-pointer-based parts in the code at all.